src/VariantGenerator/CloudflareVariantGenerator.php

```diff
@@ -171,4 +175,59 @@ private function getTempDir(): string
 
         return $dir;
     }
+
+    public function setup(VariantCollectionInterface $variantCollection): SetupResultInterface
+    {
+        $setupResult = new SetupResult($this->getName());
+
+        // TODO: Add synchronization of changes to variant
+        $this->ensureVariantsExists($variantCollection, $setupResult);
+
+        return $setupResult;
+    }
+
+    private function ensureVariantsExists(VariantCollectionInterface $variantCollection, SetupResult $setupResult): void
+    {
+        $variants = $variantCollection->getByGenerator(self::NAME);
+
+        if (empty($variants)) {
```

Comment:
Could we implement `isEmpty` or something on the variant collection?

Reply:
Good idea